classes/abstracts/ActionScheduler.php

```diff
@@ -181,21 +181,26 @@ public static function init( $plugin_file ) {
 	}
 
 	/**
-	 * Issue deprecated warning if an Action Scheduler function is called in the shutdown hook.
+	 * Check whether the AS data store has been initialized.
 	 *
 	 * @param string $function_name The name of the function being called.
+	 * @return bool
 	 */
-	public static function check_shutdown_hook( $function_name ) {
-		if ( 'shutdown' === current_filter() ) {
-			$message = sprintf(
-				/* translators: $1: open code tag, $2: function name, $3: close code tag. */
-				__( 'Action Scheduler function %1$s%2$s%3$s should not be used within the WordPress %1$sshutdown%3$s hook.', 'action-scheduler' ),
-				'<code>',
-				esc_attr( $function_name ) . '()',
-				'</code>'
-			);
-			_deprecated_hook( 'shutdown', 'Action Scheduler 3.0', 'init', $message );
+	public static function is_initialized( $function_name ) {
+		if ( did_action( 'init' ) ) {
```

Comment:
AS is also initialized when `doing_filter( 'init' )` and priority >= 2.

Reply:
@thenbrent I added a flag for this in the ActionScheduler class.

Reply:
Good approach 👍